sorting/random_pivot_quick_sort.cpp

```diff
@@ -0,0 +1,305 @@
+ *           * Initialize the pInd (partition index) from the start of the array.
+ *           * Loop through the array from start to less than end. (from start to < end).
+ *               (Inside the loop) :-
+ *                   * Check if the current element (arr[i]) is less than the pivot element in each iteration.
+ *                   * If current element in the iteration is less than the pivot element,
+ *                       then swap the elements at current index (i) and partition index (pInd) and increment the partition index by one.
+ *           * At the end of the loop, swap the pivot element with partition index element.
+ *           * Return the partition index from the function.
+ *
+ * @author [Nitin Sharma](https://github.com/foo290)
+ *
+ */
+
+
+#include <iostream>       /// for IO operations
+#include <ctime>         /// for initializing random number generator
+#include <cassert>      /// for assert
+#include <algorithm>   /// for std::is_sorted()
+#include <array>      /// for std::array
+#include <tuple>     /// for returning multiple values form a function at once
+
+
+/**
+ * @namespace sorting
+ * @brief Sorting algorithms
+ */
+namespace sorting {
+/**
+ * @brief Functions for the [Random Pivot Quick Sort](https://www.sanfoundry.com/cpp-program-implement-quick-sort-using-randomisation) implementation
+ * @namespace random_pivot_quick_sort
+ */
+    namespace random_pivot_quick_sort {
+        /**
+         * @brief Utility function to print the array
+         * @tparam T size of the array
+         * @param arr array used to print its content
+         * @returns void
+         * */
+        template<size_t T>
+        void showArray(std::array<int, T> arr) {
+            for (int i = 0; i < arr.size(); i++) {
+                std::cout << arr[i] << " ";
+            }
+            std::cout << std::endl;
+        }
+
+        /**
+         * @brief Takes the start and end indices of an array and returns a random integer between the range of those two
+         * for selecting pivot element.
+         *
+         * @param start The starting index.
+         * @param end The ending index.
+         * @returns int A random number between start and end index.
+         * */
+        int getRandomIndex(int start, int end) {
+            srand(time(nullptr)); // Initialize random number generator.
+            int randomPivotIndex = start + rand() % (end - start + 1);
+            return randomPivotIndex;
+        }
+
+        /**
+         * Nothing fancy, just a simple function to swap two elements in array.
+         * @tparam size size of the array to be passed as argument.
+         * @param arr The actual array.
+         * @param i An index of array.
+         * @param j Another index of the passed array
+         * @returns std::array<int, size> Same array with swapped elems.
+        */
+        template<size_t size>
+        std::array<int, size> swap(std::array<int, size> arr, int i, int j) {
```

Comment:
C++ already includes a built-in `swap` function (requires the `algorithm` library).